backend/test/routes/user.test.js

```diff
@@ -1,95 +1,54 @@
-      name: "test-name",
-      email: "test@example.com",
-      preferred_username: "preferred_test-name",
-      roles: [],
-      role: "admin",
-      capabilities: ["users_create_all", "users_update_all", "users_delete_all", "users_read_all"],
-    });
-  });
-
-  it("Should get a list of users when you hit /api/users", async () => {
-    userModel.findAll.mockResolvedValue([{ id: 1, name: "Alex" }]);
-    const response = await app.inject({
-      method: "GET",
-      url: "/users",
-    });
-    const responseBody = JSON.parse(response.body);
-    expect(response.statusCode).toBe(200);
-    expect(Array.isArray(responseBody.data)).toBe(true);
-    responseBody.data.forEach((userObject) => expect("id" in userObject).toBe(true));
-  });
-
-  it("Should get a single user object when you hit /api/users/:id with a valid ID", async () => {
-    userModel.findById.mockResolvedValue([{ id: 1, name: "Alex" }]);
-    const response = await app.inject({
-      method: "GET",
+const { testRoutes, routeTypes } = require("./index.js");
+
+jest.mock("../../src/models/users.js");
+
+testRoutes([
+  {
+    request: { method: "GET", url: "/users" },
+    modelFunction: userModel.findAll,
+    capabilities: ["users_read_all"],
+    type: routeTypes.General,
+  },
+  {
+    request: { method: "GET", url: "/users/1" },
+    modelFunction: userModel.findById,
+    capabilities: ["users_read_all"],
+    type: routeTypes.Specific,
+  },
+  {
+    request: {
+      method: "PUT",
       url: "/users/1",
-    });
-
-    const responseBody = JSON.parse(response.body);
-    expect(response.statusCode).toBe(200);
-    expect("id" in responseBody.data).toBe(true);
-    expect("name" in responseBody.data).toBe(true);
-  });
-});
-
-describe("Access user routes with no user", () => {
-  beforeEach(() => {
-    app = serverConfig();
-    // Mock authentication functions so we can access routes.
-    authHelper.getBearerTokenFromRequest.mockReturnValueOnce("tokenString");
-    authHelper.verifyToken.mockResolvedValue(true);
-    authHelper.getUserInfo.mockReturnValue(null);
-  });
-
-  it("Should get a list of users when you hit /api/users", async () => {
-    userModel.findAll.mockResolvedValue([{ id: 1, name: "Alex" }]);
-    const response = await app.inject({
-      method: "GET",
+      payload: {
+        email: "me@gov.bc.ca",
+      },
+    },
+    modelFunction: userModel.updateOne,
+    capabilities: ["users_create_all"],
+    type: routeTypes.Specific,
+  },
+  {
+    request: {
+      method: "POST",
       url: "/users",
-    });
-    expect(response.statusCode).toBe(401);
-  });
-
-  it("Should get a single user object when you hit /api/users/:id with a valid ID", async () => {
-    userModel.findById.mockResolvedValue([{ id: 1, name: "Alex" }]);
-    const response = await app.inject({
-      method: "GET",
-      url: "/users/1",
-    });
-    expect(response.statusCode).toBe(401);
-  });
-});
+      payload: {
```

Comment:
@mhaswell-bcgov this decision was made before you arrived, but try not to include any real names in your tests.